Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -27,21 +75,166 @@ export class PlayGameComponent implements OnInit {
    * @param chipValue
    */
   public _addToWager(chipValue: number): void {
+    if (chipValue > this.user.balance) {
+      alert('You have insufficient funds for this wager.');
+      return;
+    }
     this.wager += chipValue;
-    this.player.balance -= chipValue;
+    this.user.balance -= chipValue;
   }
 
   /**
    * Returns funds to the player's balance and resets the wager to 0.
    */
   public _clearWager(): void {
-    this.player.balance += this.wager;
+    this.user.balance += this.wager;
     this.wager = 0;
   }
 
-  public async ngOnInit(): Promise<void> {
-    this.player = await this._userApi.getUser('test');
+  /**
+  * Begins the game by allowing players to enter their name and how many decks to play with.
+  * Also determines if either the player, dealer, or both have black jack.
+  */
+  public startGame(): void {
+    if (isNaN(this.deckQty) || this.deckQty < 1) {
+      alert(`${this.deckQty} is not a valid amount of decks, please try again.`);
+      return;
+    }
+    this.deck.shuffle(this.deckQty);
+    this.player.updateHand(this.deck.deal(2));
+    this.dealer.updateHand(this.deck.deal(2));
+    this.isPregameHidden = true;
+    this.isPlayAreaHidden = false;
+
+    /*Checks for blackjacks. If anyone does, game automatically ends.*/
+    if (this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.BothBlackjack, () => this.pushOutcome());
+      return;
+    }
+    else if (this.player.hasBlackJack() && !this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.PlayerBlackjack, () => this.winOutcome());
+      return;
+    }
+    else if (!this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.DealerBlackjack, () => this.loseOutcome());
+      return;
+    }
+    this.nextMove = BlackjackOutcomes.PlayerChoice;
+    this.isGameButtonHidden = false;
+  }
+
+  /**
+   * Gives the player another card but automatically ends play if the player busts. 
+   */
+  public hit(): void {
+    this.player.updateHand(this.deck.deal(1));
+    if (this.player.hasBusted) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.DealerWin;
+      this.loseOutcome();
+      this.isPlayAgainHidden = false;
+      this.isGameButtonHidden = true;
+    }
+  }
+
+  /**
+   * Ends the players turn, reveals dealer's cards, hides game buttons, and plays
+   * the dealer's hand over two second intervals.
+   */
+  public stay(): void {
+    this.revealDealer();
+    const intervalID = setInterval(() => {
+      if (this.dealer.score > 17 || (this.dealer.score === 17 && !this.dealer.hasSoft17()) || this.dealer.hasBusted) {
```

Comment:
Think this if/else if/else block can be refactored to use your new helper method.

You call `this.revealDealer()` before `setInterval` and you set `this.isPlayAgainHidden = false` in the interval.

Inbetween those two actions you set `nextMove` and invoke one of your outcome functions.

These are all the same actions your new helper function does so it seems like this can be cleaned up a bit to be more consistent with the other portions of this component file.

Reply:
I don't think it would be better because we need to reveal the dealer even if the  game isn't over to show the dealer's hand updating live. If we use the helper function we would do another reveal dealer so I don't know if that creates redundancy or anything like that. I'll defer to you but figured since this was just a slightly different scenario I wanted to double check

Reply:
Ahh okok, I see what you're sayin. It makes sense to leave it as is in that case.